src/swsssdk/configdb.py

```diff
@@ -0,0 +1,97 @@
+"""
+SONiC ConfigDB connection module 
+
+Example:
+    config_db = ConfigDBConnector()
+    config_db.connect()
+    config_db.set_table('bgp_neighbor', '10.0.0.1', {
+        'admin_status': state
+        })
+
+TODO:
+    Will add API on keyspace subscription for incremental configuration support
+    in the incoming versions.
+"""
+
+from .dbconnector import SonicV2Connector
+
+class ConfigDBConnector(SonicV2Connector):
+
+    def __init__(self):
+        """Connect to Redis through TCP, which does not requires root.
+        """
+        super(ConfigDBConnector, self).__init__(host='127.0.0.1')
+
+    def connect(self):
+        SonicV2Connector.connect(self, self.CONFIG_DB, False)
+
+    def set_table(self, table, key, data):
+        """Write a table entry to config db.
+        Args:
+            table: Table name.
+            key: Key of table row.
+            data: Table row data in a form of dictionary {'column_key': 'value', ...}
+        """
+        client = self.redis_clients[self.CONFIG_DB]
+        _hash = '{}:{}'.format(table.upper(), key)
+        client.hmset(_hash, data)
+
+    def get_table(self, table, key):
+        """Read a table entry from config db.
+        Args:
+            table: Table name.
+            key: Key of table row.
+        Returns: 
+            Table row data in a form of dictionary {'column_key': 'value', ...}
+        """
+        _hash = '{}:{}'.format(table.upper(), key)
+        return SonicV2Connector.get_all(self, self.CONFIG_DB, _hash)
+
+    def get_table_all(self, table):
+        """Read an entire table from config db.
+        Args:
+            table: Table name.
+        Returns: 
+            Table data in a dictionary form of 
+            { 'row_key': {'column_key': 'value', ...}, ...}
+        """
+        pattern = '{}:*'.format(table.upper())
+        keys = SonicV2Connector.keys(self, self.CONFIG_DB, pattern)
+        data = {}
+        for key in keys:
```

Comment:
Return an empty dictionary. See comments.

---
In reply to: [127306170](https://github.com/Azure/sonic-py-swsssdk/pull/7#discussion_r127306170) [](ancestors = 127306170)